lib/bsearch.c

```diff
@@ -214,6 +214,40 @@ HIDDEN int bsearch_ncompare_mbox(const char *s1, int l1, const char *s2, int l2)
     }
 }
 
+HIDDEN int bsearch_uncompare_mbox(const unsigned char *s1, size_t l1,
+                                  const unsigned char *s2, size_t l2)
+{
+    ssize_t min = l1 < l2 ? l1 : l2;
+    int cmp = 0;
+
+    while (min-- > 0 && (cmp = TOCOMPARE(*s1) - TOCOMPARE(*s2)) == 0) {
+        s1++;
+        s2++;
+    }
+    if (min >= 0) {
+        return cmp;
+    } else {
+        if (l2 > l1) return -1;
+        else if (l1 > l2) return 1;
+        else return 0;
+    }
+}
+
+HIDDEN int bsearch_memtree_mbox(const unsigned char *s1, size_t l1,
+                                const unsigned char *s2, size_t l2)
+{
+    size_t min = l1 < l2 ? l1 : l2;
+    int cmp = 0;
+
+    while ((min-- >0) &&
+           (cmp = (TOCOMPARE(*s1) - TOCOMPARE(*s2))) == 0) {
+        s1++;
+        s2++;
+    }
+
+    return cmp;
```

Comment:
Does this suffer from the same bug as was fixed in cmpstringp_mbox by aff48467b2?

Reply:
Good find! It probably is. Will fix that.